pkg/controller/admissionchecks/provisioning/controller.go

```diff
@@ -343,6 +339,23 @@ func (c *Controller) remainingTimeToRetry(pr *autoscaling.ProvisioningRequest, f
 	return backoffDuration - timeElapsedSinceLastFailure
 }
 
+func (c *Controller) handleCreationError(ctx context.Context, wl *kueue.Workload, ac *kueue.AdmissionCheckState, msg string, err error) error {
+	msg = api.TruncateEventMessage(fmt.Sprintf("%s: %v", msg, err))
+	c.record.Eventf(wl, corev1.EventTypeWarning, "FailedCreate", msg)
```

Comment:
Can we create this event before calling the function and thus make the function more versatile? It doesn't need to narrow its scope only to creation failure but be more generic

Reply:
Good point. Thanks! Done.